_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: post
+title: "Accessible error messages in forms"
+permalink: /blog/2025/accessible-error-messages-forms/
+author: Rian Rietveld
+categories: forms
+parent: Articles
+excerpt: "When you put care into preventing errors and clearly indicating when something goes wrong, users are much more likely to successfully submit a form."
+date: 2025-08-16
+---
+
+When you put care into preventing errors and clearly indicating when something goes wrong, users are much more likely to successfully submit a form.
+{: .fs-6 .fw-300 }
+
+In this article, we’ll walk step-by-step through how to prevent errors, indicate them clearly, and offer help with error messages in forms for different types of users.
+
+Ask yourself the following questions:
+
+* What do I really need to know?
+* How will I ask for this information?
+* What information can I provide in advance to help the user?
+* How to show which fields are required?
+* When do I check for errors?
+* How do I indicate that an answer is incomplete or filled out incorrectly?
+* What makes for a helpful error message?
+* How do I confirm that a form has been submitted successfully?
+* How can I offer help when a user gets stuck?
+
+No one enjoys filling out forms, and all the help you can give is useful. Keep that goal in mind. You want to learn something from your user, or your user wants to tell you something. Make that process as easy as possible.
+
+## What do I really need to know?
+
+Handling errors starts with the questions you ask. Some questions are difficult to answer. Is it really essential to know whether someone is male or female? Do you only want to contact them by phone? Perhaps the user wants to provide a different answer or doesn't want to answer, can’t choose an option and then gives up.
+
+Only ask for information you truly need in order to process the request. The fewer questions you ask, the lower the barrier for filling out the form.
+
+Be very clear about why you need sensitive personal information, such as a citizen service number, race, religion or medical data. Check your local legislation if you are allowed to request such private information in an open online form.
+
+{: .note }
+In the Netherlands there are strict rules about [when to ask for a citizen service number](https://www.rijksoverheid.nl/onderwerpen/privacy-en-persoonsgegevens/vraag-en-antwoord/welke-organisaties-mogen-mijn-burgerservicenummer-bsn-gebruiken#:~:text=Alle%20overheidsorganisaties%20mogen%20gebruik%20maken%20van%20uw%20burgerservicenummer%20(BSN)).
+
+
+## How will I ask for this information?
+
+Will you design new custom input fields or use recognizable, familiar patterns? Please don’t reinvent the web for something as essential as form fields. As Heydon Pickering explains in his talk [Get Your Priorities Straight](https://www.youtube.com/watch?v=ediHVy0869c): “Real people aren't looking to be delighted. People want to get the task done and get on with their lives".
```

Comment:
I think this needs another paragraph first. I didn't immediately connect the topic of custom inputs to the question in the heading.

E.g.

"The input mechanism you use to ask for information has a profound impact on your users."

then

"Should you design new custom input fields or use recognizable, familiar patterns?"

(Change to 'should' instead of 'will'.)